app/main.py

```diff
@@ -1,25 +1,84 @@
 from abc import ABC
+from typing import Type, Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self,
+                 min_amount: int,
+                 max_amount: int
+                 ) -> None:
+
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self,
+                     owner: Type[object],
+                     name: str
+                     ) -> None:
+        self.public_name = name
+        self.protected_name = "_" + name
+
+    def __get__(self,
+                instance: object,
+                owner: Type[object]) -> Any:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object,
+                value: Any) -> Any:
+        if self.min_amount <= value <= self.max_amount:
+            setattr(instance, self.protected_name, value)
```

Comment:
better to raise exception if value is incorrect
after that you can catch this exception in `can_access`

Reply:
Got it! Thanks!